src/jsMain/kotlin/process/phonemes/PhonemesMapping.kt

```diff
@@ -22,7 +22,9 @@ data class PhonemesMappingRequest(
         val from = line.substringBefore("=").trim()
         val to = line.substringAfter("=").trim()
         from to to
-    }.sortedByDescending { it.first.split(" ").size }
+    }
+        .sortedByDescending { it.first.length }
+        .sortedByDescending { it.first.split(" ").size }
```

Comment:
I think`.sortedByDescending { it.first.split(" ").size }` should come first?

So that we have:
- The line with more phonemes are prioritized
- If the phonemes counts are the same, longer phonemes are prioritized.

We might have entries
- `a b c=...`
- `asdasdasd a=...` 
and when we have a note with `asdasdasd a b c`, the entries are conflicted.

Reply:
The sorting order is correct, since the phoneme count sort is done last the resulting map will prioritize them.

The problem appears to be in `Note.replacePhonemes()` since it moves left to right to find groups of phonemes instead of being able to match the largest groups in arbitrary positions. I'll push a test showing this, since I'm not sure how to fix it yet.

Reply:
Nice catch!
I pushed a new commit: 7ed1431c472bd64fc703bc089fff987a15ac9134
that fixes the process and included your test cases. Please merge it to your PR.

Reply:
Oh, the sorting was wrong, added one more commit.